addons/stock/models/stock_move.py

```diff
@@ -270,14 +278,8 @@ def _compute_is_quantity_done_editable(self):
         for move in self:
             if not move.product_id:
                 move.is_quantity_done_editable = False
-            elif not move.picking_id.immediate_transfer and move.picking_id.state == 'draft':
-                move.is_quantity_done_editable = False
             elif move.picking_id.is_locked and move.state in ('done', 'cancel'):
```

Comment:
```
elif move.state in ('done', 'cancel') and (
          move.picking_id.is_locked
          or move.show_details_visible
          or move.show_operations
     ):
    move.is_quantity_done_editable = False
```
This change will correctly cover the cases we discussed yesterday (I probably messed up the spacing of that code snippet).

It looks like weird behavior will happen if you change the `quantity_done` after the picking is done and either show_details_visible or is_quantity_done_editable is true